src/server/mod.rs

```diff
@@ -235,7 +235,7 @@ impl<'a> RequestAction<'a> for InitializeRequest {
                 signature_help_provider: None,
             }
         };
-        out.success(id, &result);
```

Comment:
We must respond to the `InitializeRequest` with a `InitializeResponse`, so I don't think we can just omit a response here. Now that I look at it @nrc, shouldn't the `Response` return the optional Notification instead of explicitly calling `success` on passed `Output`? I didn't take an in-depth look at how traits changed the implementation details of the server, I have some catching up to do.

Reply:
Usually yes, that is what we should do, but in this case we want to send the response before we do the work below, so we can't return the result at the end of the function body.

Reply:
I think I'll close this PR, as it doesn't sound like the right way to solve this.